packages/main/src/Menu.ts

```diff
@@ -359,328 +246,126 @@ class Menu extends UI5Element {
 		return isPhone();
 	}
 
-	get isSubMenuOpened() {
-		return this._parentMenuItem && this._popover?.isOpen();
-	}
-
-	get menuHeaderTextPhone() {
-		return this._parentMenuItem ? this._parentMenuItem.text : this.headerText;
+	get _popover() {
+		return this.shadowRoot!.querySelector<ResponsivePopover>("[ui5-responsive-popover]")!;
 	}
 
 	onBeforeRendering() {
-		this._prepareCurrentItems(this.items);
-
-		const itemsWithChildren = this.itemsWithChildren;
-		const itemsWithIcon = this.itemsWithIcon;
-
-		this._currentItems.forEach(item => {
-			item.item._siblingsWithChildren = itemsWithChildren;
-			item.item._siblingsWithIcon = itemsWithIcon;
-			const subMenu = item.item._subMenu;
-			const menuItem = item.item;
-			if (subMenu && subMenu.loading) {
-				subMenu.innerHTML = "";
-				const fragment = this._clonedItemsFragment(menuItem);
-				subMenu.appendChild(fragment);
-			}
-
-			if (subMenu) {
-				subMenu.loading = item.item.loading;
-				subMenu.loadingDelay = item.item.loadingDelay;
-			}
-		});
-	}
-
-	onAfterRendering() {
-		if (!this.opener) {
-			return;
-		}
-
-		if (this.open) {
-			const opener = this.getOpener();
-			if (opener && !this.isSubMenuOpened) {
-				this.showAt(opener);
-			}
-		} else {
-			this.close();
-		}
-	}
+		const siblingsWithIcon = this.items.some(item => !!item.icon);
 
-	/**
-	 * Shows the Menu near the opener element.
-	 * @param opener the element that the popover is shown at
-	 * @public
-	 */
-	async showAt(opener: HTMLElement): Promise<void> {
-		if (!this._isSubMenu) {
-			this._parentMenuItem = undefined;
-			this._opener = undefined;
+		if (siblingsWithIcon) {
```

Comment:
what if there is one sibling with icon, and at some moment the icon is removed? Nobody will reset other items' _siblingsWithIcon property... Maybe will be better to do:

```
this.items.forEach(item => {
	item._siblingsWithIcon = siblingsWithIcon;
});
```

without the prior IF?